### ISSUE TITLE
Doesn't differentiate between `module-name` and `@namespace/module-name`

### ISSUE THREAD

**Issue by mikestopcontinues** [benmosher/eslint-plugin-import#1747]
I'm working in a monorepo with a package named `@arc/dotenv` which requires `dotenv`. This triggers `import/no-self-import`. My suspicion is that it's not taking the namespace into account. Another possibility: the package lives at `<root>/tools/dotenv`, so perhaps the issue comes up after the module gets resolved?

**Comment by ljharb** [comment]
Can you share the exact code it's warning on?

**Comment by mikestopcontinues** [comment]
Sure!

This is in `<root>/tools/dotenv/index.js`. The module is named `@arc/dotenv`, `"main": "index.js"`.

```js
// import

const fs = require('fs');
const path = require('path');

const expand = require('dotenv-expand');

const dotenv = require('dotenv'); // this triggers`import/no-self-import`
```

<img width="709" alt="Screen Shot 2020-05-04 at 4 10 59 PM" src="https://user-images.githubusercontent.com/150434/81014097-d95c6f00-8e21-11ea-9d49-31b850e51d2f.png">

**Comment by ljharb** [comment]
and the package.json in `/tools/dotenv/package.json`'s `name` field?

**Comment by mikestopcontinues** [comment]
Sorry, I should've included that.

```json
{
  "name": "@arc/dotenv",
  "version": "0.0.0",
  "private": true,
  "main": "index.js",
  "devDependencies": {
    "dotenv": "8.2.0",
    "dotenv-expand": "5.1.0"
  }
}

```
